### PR TITLE
Funky fresh: Additional adjustments to repo setup for fresh machines

### PR DESCRIPTION
**Pull request by Shadowfiend**
I went through the `local-setup` installation process on a fresh machine, which found a few more gaps I'd missed in our scripts:
- An explicit choice of Node 12 for fresh setups. Node 12 is the LTS version, 11 is no longer supported.
- A few unsupported API declarations in run-geth for latest geth.
- An issue with relay difficulty for the local regtest network.

Those changes are reflected here.

### CODE REVIEW

File: relay-genesis/update-mock-relay.js [hunk 1]
```diff
@@ -15,9 +15,9 @@ module.exports = async function() {
 
         const mockRelay = await MockRelay.deployed()
 
-        await mockRelay.setCurrentEpochDifficulty(difficulty)
-        await mockRelay.setPrevEpochDifficulty(difficulty)
+        await mockRelay.setCurrentEpochDifficulty(Math.min(difficulty, 1))
+        await mockRelay.setPrevEpochDifficulty(Math.min(difficulty, 1))
```
Comment (lukasz-zimnoch) [inline relay-genesis/update-mock-relay.js:19]: Why are we changing this? The `MockRelay` contract has `1` as default current and previous difficulty. IIRC, the regtest network has a difficulty less than `1` so actually this script sets the values in the contract to `0` and due to that, allows transaction proofs to be accepted successfully. Am I missing something?

File: docs/manual-setup.adoc [hunk 2]
```diff
@@ -20,7 +20,7 @@ Regardless you use `macos-setup.sh` or not, you should also install:
 
 == Node.js version notice
 
-Please use Node `v11.15.0` for everything but E2E test script.
+Please use Node `v12.18.3` for everything but E2E test script.
```
Comment (lukasz-zimnoch) [inline docs/manual-setup.adoc:23]: Can we also align the automatic setup to that version? Especially, the `deployments/local-setup-environment/provisioning/install-nodejs.sh` script.

File: run-geth.sh [hunk 3]
```diff
@@ -16,8 +16,8 @@ printf "${LOG_START}Starting geth...${LOG_END}"
 geth --port 0 --networkid 1101 --identity "somerandomidentity" \
     --ws --wsaddr "127.0.0.1" --wsport "8546" --wsorigins "*" \
     --rpc --rpcport "8545" --rpcaddr "127.0.0.1" --rpccorsdomain "*" \
-    --rpcapi "db,ssh,miner,admin,eth,net,web3,personal,debug" \
-    --wsapi "db,ssh,miner,admin,eth,net,web3,personal,debug" \
+    --rpcapi "miner,admin,eth,net,web3,personal,debug" \
+    --wsapi "miner,admin,eth,net,web3,personal,debug" \
```
Comment (lukasz-zimnoch) [inline run-geth.sh:20]: Are we also changing the recommended `geth` version? Currently we recommend `1.9.9-stable` in the `manual-setup.adoc` and use the same in `deployments/local-setup-environment/provisioning/install-geth.sh` for automatic setup.

**Review by lukasz-zimnoch — CHANGES_REQUESTED** [review]

File: relay-genesis/update-mock-relay.js (hunk 1 above)
Comment (Shadowfiend) [reply relay-genesis/update-mock-relay.js:19]: This should definitely be Math.max, not min. My goal here was to make sure running this did not break the mock relay if someone set it while running regtest. Does that make sense?

File: run-geth.sh (hunk 3 above)
Comment (Shadowfiend) [reply run-geth.sh:20]: We can (I'm running latest, and the macos-setup script installs latest anyway). I'm fine either way to be honest---this makes it work on latest and shouldn't break 1.9.9 so it also feels fine to leave it for now.

Comment (lukasz-zimnoch) [reply run-geth.sh:20]: Ok, good!

File: relay-genesis/update-mock-relay.js (hunk 1 above)
Comment (lukasz-zimnoch) [reply relay-genesis/update-mock-relay.js:19]: Hmm, but if we use `Math.max` we can never set the difficulty to a value less than `1` which is needed when we use`regtest`.

Comment (Shadowfiend) [reply relay-genesis/update-mock-relay.js:19]: regtest has a difficulty less than 1? 🤔 The contracts return a minimum difficulty of 0 for SPV proofs, so I don't think we can verify a 0-difficulty transaction. I might be misunderstanding a few bits here.

Comment (lukasz-zimnoch) [reply relay-genesis/update-mock-relay.js:19]: The [`getdifficulty`](https://bitcoincore.org/en/doc/0.20.0/rpc/blockchain/getdifficulty/) method of the bitcoin RPC returns something between `0` and `1` when executed against a `regtest` client. But, solidity just cuts the decimal part and we have a `0` set in the `MockRelay` contract.

Comment (Shadowfiend) [reply relay-genesis/update-mock-relay.js:19]: Mmkay, so I'm just going to revert this for now and we'll see if I can reproduce my prior issues.

**Comment by nkuba** [comment]
I let myself merge the main branch and resolve conflicts. Additionally I updated the node version to 14 as this is the version that we confirmed recently to work fine through the whole setup cycle.

**Review by nkuba — APPROVED** [review]
LGTM, but since I contributed to this PR I'm leaving the final merge decision to y'all.

**Comment by beaushinkle** [comment]
This gets rid of confusing cruft that cost me a bunch of time during my own onboarding :100:

**Comment by Shadowfiend** [comment]
@beaushinkle [appuyez le bouton](https://translate.google.com/?sl=fr&tl=en&text=appuyez%20le%20bouton!&op=translate)!

**Comment by beaushinkle** [comment]
I can't!

image

**Comment by Shadowfiend** [comment]
Ohp. Fixed.

**Review by lukasz-zimnoch — APPROVED** [review]

**Pull request merged**